src/main/java/net/fabricmc/loom/configuration/providers/minecraft/mapped/MappedMinecraftProvider.java

```diff
@@ -42,6 +42,21 @@ interface ProviderImpl extends MappedMinecraftProvider {
 	}
 
 	interface Merged extends ProviderImpl {
+		/*
+		 * For legacy versions of Minecraft, the client and server
+		 * jars are remapped separately and then merged together.
+		 * For this the remapped client and server jars must be
+		 * available here as well, as an intermediate step.
+		 */
+
+		default MinecraftJar getClientJar() {
+			return new MinecraftJar.Client(getJar(MinecraftJar.Type.CLIENT));
+		}
+
+		default MinecraftJar getServerJar() {
+			return new MinecraftJar.Server(getJar(MinecraftJar.Type.SERVER));
+		}
```

Comment:
Im having 2nd thoughts on if its a good idea or not to use the same provider for this. The idea was for it to be clear what jars are going to be present, in this case its not that clear. Let me know if you want me to have a look at this.

Reply:
I'd appreciate it if you looked into it. The reason I did it this way is because these methods are used in all the impls of these providers, e.g. in `NamedMinecraftProvider`:
```java
		@Override
		public List<RemappedJars> getRemappedJars() {
			return minecraftProvider.canMergeJars()
				? List.of(
					new RemappedJars(minecraftProvider.getMergedJar(), getMergedJar(), MappingsNamespace.OFFICIAL)
				) : List.of(
					new RemappedJars(minecraftProvider.getClientJar(), getClientJar(), MappingsNamespace.CLIENT_OFFICIAL),
					new RemappedJars(minecraftProvider.getServerJar(), getServerJar(), MappingsNamespace.SERVER_OFFICIAL)
				);
		}
```
I figured it would make more sense to have these 'mapped client' and 'mapped server' jars in the mapped minecraft providers as opposed to the base minecraft provider.

Reply:
I tried moving these methods out of `ProviderImpl` and into the specific `MergedImpl`s where they're used. Is that a better solution?